src/DashboardService.Data/TaskRepository.cs

```diff
@@ -18,28 +21,114 @@ public TaskRepository(IDataProvider provider)
     _provider = provider;
   }
   
-  public Task<Guid?> CreateAsync(DbTask dbDepartment)
+  public async Task<Guid> CreateAsync(DbTask dbTask)
   {
-    throw new NotImplementedException();
+    _provider.Tasks.Add(dbTask);
+    await _provider.SaveAsync();
+
+    return dbTask.Id;
   }
 
-  public Task<List<DbTask>> GetAllAsync(GetTasksFilter filter)
+  public async Task<(List<DbTask> dbTasks, int totalCount)> GetAllAsync(GetTasksFilter filter, CancellationToken ct)
   {
-    throw new NotImplementedException();
+    IQueryable<DbTask> dbTasks = _provider.Tasks.Where(t => !filter.GroupId.HasValue || t.GroupId == filter.GroupId).AsNoTracking();
+
+    if (filter.BoardId.HasValue)
+    {
+      DbBoard dbBoard = await _provider.Boards
+        .Include(b => b.Groups)
+        .FirstOrDefaultAsync(b => b.Id == filter.BoardId, ct);
+      
+      dbTasks = dbTasks
+        .Include(t => t.Group)
+        .Where(t => dbBoard.Groups.FirstOrDefault(t.Group) != default);
+    }
+    
+    dbTasks = dbTasks.Where(t => !filter.PriorityId.HasValue || t.PriorityId == filter.PriorityId).AsNoTracking();
+    dbTasks = dbTasks.Where(t => !filter.TaskTypeId.HasValue || t.TaskTypeId == filter.TaskTypeId).AsNoTracking();
+    dbTasks = dbTasks.Where(t => !filter.DeadlineAtUtc.HasValue || t.DeadlineAtUtc < filter.DeadlineAtUtc).AsNoTracking();
+    dbTasks = dbTasks.Where(t => !filter.CreatedBy.HasValue || t.CreatedBy == filter.CreatedBy).AsNoTracking();
+
+    if (filter.IsAscendingSort.HasValue)
+    {
+      dbTasks = filter.IsAscendingSort.Value
+        ? dbTasks.OrderBy(d => d.Name)
+        : dbTasks.OrderByDescending(d => d.Name);
+    }
+
+    if (!string.IsNullOrWhiteSpace(filter.NameIncludeSubstring))
+    {
+      dbTasks = dbTasks.Where(d => d.Name.Contains(filter.NameIncludeSubstring));
+    }
+
+    return (
+      await dbTasks
+        .Skip(filter.SkipCount)
+        .Take(filter.TakeCount)
+        .ToListAsync(ct),
+      await dbTasks.CountAsync(ct));
   }
 
-  public Task<DbTask> GetAsync(Guid id, GetTaskFilter filter)
+  public async Task<DbTask> GetAsync(Guid id, GetTaskFilter filter, CancellationToken ct)
   {
-    throw new NotImplementedException();
+    return await IncludeRelatedEntities(filter, _provider.Tasks).
+      AsNoTracking().
```

Comment:
In other cases you make '.' before line? not after

Reply:
fixed